src/library/scala/collection/immutable/Range.scala

```diff
@@ -165,6 +165,41 @@ sealed abstract class Range(
     }
   }
 
+  override final def indexOf[@sp(Int) B >: Int](elem: B, from: Int = 0): Int =
+    elem match {
+      case i: Int =>
+        val pos = posOf(i)
+        if (pos >= from) pos else -1
+      case _ => super.indexOf(elem, from)
+    }
+
+  override final def lastIndexOf[@sp(Int) B >: Int](elem: B, end: Int = -1): Int =
```

Comment:
the default for end should be `length - 1`

otherwise the result would always be -1 when not supplying an `end` index 

as per SeqOps

https://github.com/scala/scala/blob/b78715242877fc623744f3d06d87e37d576945e9/src/library/scala/collection/Seq.scala

Reply:
oops, good catch.

[ ] that